src/summaries/control_flow.yaml

```diff
@@ -0,0 +1,739 @@
+    - Unless you provide an else expression, if the condition is false, the program will skip the if block and move on.
+    The following output should be seen after running this code:
+    ```rust
+    $ cargo run
+     Compiling branches v0.1.0 (file:///projects/branches)
+     Finished dev [unoptimized + debuginfo] target(s) in 0.31s
+      Running `target/debug/branches`
+    condition was true
+    ```
+    - If we change the `number` to 17 we get a `condition was false` because it is not less than 5. It is a very straight forward statement.
+- title: Multiple if conditions - how to evaluate them?
+  shortTitle: Nested if statement
+  key: if_statement
+  summary: The two logical operators && and || allow us to evaluate more than one condition in an if statement.
+  details: |
+    - Using conditional AND, you can test whether one condition and another are true. Before the code can be executed, both conditions must be met.
+    - When AND is used to evaluate multiple conditions, the && operator is used to separate them.
+    ```rust
+    fn main() {
+     let a = 5;
+     if a > 0 && a < 10 {
+         println!("Both conditions are true");
+     }
+    }
+    ```
+    - Due to the fact that both of our conditions are true in the example above, the code inside the if code block is executed.
+    - It tests whether one condition or another is true by using the conditional OR. It is not necessary for both conditions to be true in order for the code to run.
+    - The || operator is used to separate multiple conditions with OR.
+    ```rust
+    fn main() {
+      let a = 5;
+      if a > 5 || a < 10 {
+         println!("One of the conditions are true");
+     }
+     }
+    ```
+- title: How to nest if, else if and else statements?
+  shortTitle: Nested if continues
+  key: Nested_if_else_statement
+  details: |
+    - Nesting is the process of placing if statements inside other if statements. 
+    - By performing the evaluations hierarchically, the compiler will start at the outer if statement and work inwards.
+    - We nest an if statement by writing it inside the execution code block of another if statement.
+    ```rust
+    fn main() {
+     let number = 5;
+     if number < 10 {
+         // if outer condition proves
+         // true evaluate the inner if
+         if number > 0 {
+             println!("Inner if statement");
+         }
+      }
+    }
+    ```
+    - In the example above, our outer if statement condition proves true. 
+    - The compiler then evaluates the inner if statement and executes its code block if its condition proves true.
+    - Nested if statements are used when we want to evaluate conditions only if other conditions are true.
+    - Note that if you find yourself nesting more than 3 levels deep, you should consider finding a different method. Perhaps a loop.
+- title: The conditional match statement
+  shortTitle: Match statement
+  key: Match_statement
```

Comment:
please update all the keys to small case and hyphen separated

Reply:
I also edited subtopic in question and reading to match it